source/Cosmos.HAL2/PIT.cs

```diff
@@ -1,280 +1,341 @@
+
+            #region (deprecated)
+
+            [Obsolete($"Use the {nameof(Recurring)} property instead.")]
+            public bool Recuring => Recurring;
+
+            #endregion
+        }
+
+        public const uint PITFrequency = 1193180;
+        public const uint PITDelayNS = 838;
+
+        public bool T0RateGen = false;
+
+        protected Core.IOGroup.PIT IO = Core.Global.BaseIOGroups.PIT;
+        private List<PITTimer> activeHandlers = new();
+        private ushort _T0Countdown = 65535;
+        private ushort _T2Countdown = 65535;
+        private int timerCounter = 0;
+        private bool waitSignaled = false;
+
+        public PIT()
+        {
+            INTs.SetIrqHandler(0x00, HandleIRQ);
+            T0Countdown = 65535;
+        }
+
+        public ushort T0Countdown
+        {
+            get => _T0Countdown;
+            set {
+                _T0Countdown = value;
+
+                IO.Command.Byte = (byte)(T0RateGen ? 0x34 : 0x30);
+                IO.Data0.Byte = (byte)(value & 0xFF);
+                IO.Data0.Byte = (byte)(value >> 8);
+            }
+        }
+
+        public uint T0Frequency
+        {
+            get => PITFrequency / (uint)_T0Countdown;
+            set {
+                if (value < 19 || value > 1193180) {
+                    throw new ArgumentException("Frequency must be between 19 and 1193180!");
+                }
+
+                T0Countdown = (ushort)(PITFrequency / value);
+            }
+        }
+
+        public uint T0DelayNS
+        {
+            get => PITDelayNS * _T0Countdown;
+            set {
+                if (value > 54918330) {
+                    throw new ArgumentException("Delay must be no greater that 54918330");
+                }
+
+                T0Countdown = (ushort)(value / PITDelayNS);
+            }
+        }
+
+        public ushort T2Countdown
+        {
+            get => _T2Countdown;
+            set
+            {
+                _T2Countdown = value;
+
+                IO.Command.Byte = 0xB6;
+                IO.Data0.Byte = (byte)(value & 0xFF);
+                IO.Data0.Byte = (byte)(value >> 8);
+            }
+        }
+
+        public uint T2Frequency
+        {
+            get => PITFrequency / ((uint)_T2Countdown);
+            set
+            {
+                if (value < 19 || value > 1193180)
+                {
+                    throw new ArgumentException("Frequency must be between 19 and 1193180!");
+                }
+
+                T2Countdown = (ushort)(PITFrequency / value);
+            }
+        }
+
+        public uint T2DelayNS
+        {
+            get => (PITDelayNS * _T2Countdown);
+            set
+            {
+                if (value > 54918330) {
+                    throw new ArgumentException("Delay must be no greater than 54918330");
+                }
+
+                T2Countdown = (ushort)(value / PITDelayNS);
+            }
+        }
+
+        public void EnableSound()
```

Comment:
Can we also deprecate all the sound methods?

Reply:
Sure, will mark with `[Obsolete]` as to not break any old kernels